ee/tabby-db/src/email_service_credential.rs

```diff
@@ -47,13 +47,14 @@ impl DbConn {
         Ok(self
             .conn
             .call(move |c| {
+                let transaction = c.transaction()?;
                  let smtp_password = match smtp_password {
                     Some(pass) => pass,
                     None => {
```

Comment:
please add a unit test for this

Reply:
If you meant add a unit test to ensure the transaction is atomic, I'm not sure how to do that. I added another test case covering updating everything other than the password, which wasn't covered before. Let me know if I misunderstood.

Reply:
right, i mean to add a test for calling `update_email_service_credential` with `smtp_password=None`, saw you've added it.